### PR TITLE
Update/0.14.2_b2

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update dependencies to what is actually needed
- Update tests and about section

Reason: missing on some archs, needed to build [scikit-bio](https://github.com/AnacondaRecipes/scikit-bio-feedstock/pull/1)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,33 +10,46 @@ source:
...
+  number: 2
   script: '{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv '
 
 requirements:
   build:
     - {{ compiler('c') }}
   host:
-    - cython >=0.23
-    - numpy
+    - cython
```
Comment (marcoesters) [inline recipe/meta.yaml:24]: `cython` is still contrained: https://github.com/daleroberts/hdmedians/blob/v0.14.2/pyproject.toml#L2

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,33 +10,46 @@ source:
...
 test:
   imports:
     - hdmedians
+  requires:
+    - pip
+    - nose >=1.0
+  source_files:
+    - hdmedians/tests
+  commands:
+    # the pypi package is misconfigured and lists cython and nose as run dependencies.
```
Comment (marcoesters) [inline recipe/meta.yaml]: There appear to be two conflicting requirement lists:

https://github.com/daleroberts/hdmedians/blob/v0.14.2/setup.py#L17-L18
https://github.com/daleroberts/hdmedians/blob/v0.14.2/pyproject.toml

I'm not sure how this is prioritized, but maybe those lines need to be patched?

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:24]: Given we are at 0.29, I think we can drop `>=0.23`

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: that works too, I added a patch

**Review by marcoesters — APPROVED** [review]

**Review by ELundby45 — APPROVED** [review]
LGTM! 👍

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,35 +8,51 @@ package:
...
   imports:
     - hdmedians
+  requires:
+    - pip
+    - nose >=1.0
+  source_files:
+    - hdmedians/tests
+  commands:
+    - pip check
+    - pip check
```
Comment (marcoesters) [inline recipe/meta.yaml]:
```suggestion
```

**Pull request merged**